components/Profile/DetailUpdateModal.js

```diff
@@ -0,0 +1,118 @@
+/* eslint-disable prettier/prettier */
+//import libraries
+import React from 'react';
+import { View, Text, StyleSheet, Modal, TextInput, Alert, Button} from 'react-native';
+import color from '../../constants/color';
+
+/**
+ *
+ * @param {string} prop.toggleDetailModal function that toggles visible or invisible
+ * @param {string} prop.modalVisible Actual Boolean being toggled in parent
+ * @param {string} prop.originalDetail the original detail (username)
+ * @param {Boolean} prop.onConfirm what to do when the user clicks save
+ *
+ */
+// create a component
+const DetailUpdateModal = ({toggleDetailModal, modalVisible, originalDetail, onConfirm}) => {
+  const [detail, setDetail] = React.useState('');
+  const [errorMessage, setErrorMessage] = React.useState('');
+  const legalChars = /^[a-zA-Z0-9_-]*$/; // allow letters, numbers, and underscores
+
+  const updateUserDetails = () => {
+    if (detail.length < 5){
+      setErrorMessage('The username must be longer than 5 characters');
+      return;
+    } else if (!legalChars.test(detail)){
+      setErrorMessage('The username can only contain letters, numbers and underscores');
+      return;
+    } else if (originalDetail === detail){
+      setErrorMessage('');
+      setDetail('');
+      toggleDetailModal();
+      console.log('username unchanged ');
+      return;
+    }
+
+    //if all is good
+    onConfirm(detail);
+    setDetail('');
+    toggleDetailModal();
+    console.log('new username is ', detail);
+  };
+  const cancelChange = () => {
+    toggleDetailModal();
+    setDetail('');
+  };
+
+return (
+  <Modal
+    animationType="slide"
+    visible={modalVisible}
+    presentationStyle="fullScreen"
+    onRequestClose={() => {
+      toggleDetailModal();
+      Alert.alert('Username change aborted');
+    }}
+  >
+    <View style={styles.container}>
+      <Text style={styles.errorMessage} >{errorMessage}</Text>
+      <TextInput style={styles.input}
+        onChangeText={(text) => setDetail(text)}
```

Comment:
This could be simply `onChangeText={setDetail}`

Reply:
I see.
Isn't that just a short form of writing the same thing? Or what is the difference?
Is it just for the appearance or there is some difference? performance benefits

Reply:
It's both a shorter form and more performant. `(text) => setDetail(text)` is a new function every time the `onChangeText` gets called, while `setDetail` will be referenced if using the shorter form (no new function initialization)